pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,148 @@
+    then builtins.concatStringsSep "/" (lib.drop 1 (lib.splitString "/" (lib.removePrefix "${builtins.storeDir}/" (toString depsPath))))
+    # if this is an absolute path, just use that path
+    else if lib.hasPrefix "/" (toString depsPath)
+    then toString depsPath
+    # otherwise, use a path relative to the package
+    else ''$(nix-instantiate --eval -E "dirOf (${code}).meta.position" | tr -d '"')/${depsPath}''
+  }"
+
+  nix-build --no-out-link -E "${code'}" > /dev/null
+
+  pushd "$(${coreutils}/bin/mktemp -d)"
+  MITM_CACHE_DIR="$PWD"
+  trap "${coreutils}/bin/rm -rf '$MITM_CACHE_DIR'" SIGINT SIGTERM ERR EXIT
+  ${openssl}/bin/openssl genrsa -out ca.key 2048
+  ${openssl}/bin/openssl req -x509 -new -nodes -key ca.key -sha256 -days 1 -out ca.cer -subj "/C=AL/ST=a/L=a/O=a/OU=a/CN=example.org"
+  export MITM_CACHE_HOST=127.0.0.1
+  export MITM_CACHE_PORT="''${mitmCachePort:-$(${python3.pkgs.ephemeral-port-reserve}/bin/ephemeral-port-reserve "$MITM_CACHE_HOST")}"
+  export MITM_CACHE_ADDRESS="$MITM_CACHE_HOST:$MITM_CACHE_PORT"
+  # forget all redirects - this makes the lockfiles predictable
+  # not only does this strip CDN URLs, but it also improves security - since the redirects aren't
+  # stored in the lockfile, a malicious actor can't change the redirect URL stored in the lockfile
+  ${mitm-cache}/bin/mitm-cache \
+    -l"$MITM_CACHE_ADDRESS" \
+    record \
+    --reject '\.(md5|sha(1|256|512:?):?)$' \
+    --forget-redirects-from '.*' \
+    --record-text '/maven-metadata\.xml$' >/dev/null 2>/dev/null &
+  MITM_CACHE_PID="$!"
+  # wait for mitm-cache to fully start
+  for i in {0..20}; do
+    ${procps}/bin/ps -p "$MITM_CACHE_PID" >/dev/null || (echo "Failed to start mitm-cache" && exit 1)
+    ${curl}/bin/curl "$MITM_CACHE_ADDRESS" && break
+    [[ "$i" -eq 20 ]] && (echo "Failed to start mitm-cache" && exit 1)
+    sleep 0.5
+  done
+  trap "kill '$MITM_CACHE_PID'" SIGINT SIGTERM ERR EXIT
+  export MITM_CACHE_CERT_DIR="$PWD"
+  export MITM_CACHE_CA="$MITM_CACHE_CERT_DIR/ca.cer"
+  popd
+  useBwrap="''${USE_NSJAIL:-${toString useBwrap}}"
+  nixShell="$(${coreutils}/bin/mktemp --suffix=.nix)"
+  trap "${coreutils}/bin/rm -f '$nixShell'" SIGINT SIGTERM ERR EXIT
+  echo "${code'}" > "$nixShell"
+  if [ -n "$useBwrap" ]; then
+    # bwrap isn't necessary, it's only used to prevent messy build scripts from touching ~
+    ${bubblewrap}/bin/bwrap \
+      --unshare-all --share-net --clearenv --chdir / --setenv HOME /homeless-shelter \
+      --tmpfs /home --bind /tmp /tmp --ro-bind /nix /nix --ro-bind /run /run --proc /proc --dev /dev  \
+      --ro-bind ${toString path} ${toString path} --ro-bind "$nixShell" "$nixShell" --bind "$MITM_CACHE_CERT_DIR" "$MITM_CACHE_CERT_DIR" \
+      ${builtins.concatStringsSep " " (map (x: "--setenv ${x} \"\$${x}\"") keepSandbox)} \
+      --setenv NIX_PATH nixpkgs=${toString path} ${bwrapFlags} \
+      -- $(${pkgs.which}/bin/which nix-shell) --pure --run ${gradleScript} ${nixShellKeep} "$nixShell"
```

Comment:
if `nix-shell` isn't in /nix or /run, this won't work (but it's possible to disable the sandbox anyway)

Reply:
Why not simply `${lib.getBin pkgs.nix}/bin/nix-shell}`?

Reply:
Personally, my system is very sensitive towards Nix versions, as I use Nix plugins, which are only binary compatible with a single Nix version at a time, so I think it's a good idea to use the system Nix version whenever possible